src/components/tl/ucp/scatter/scatter_knomial.c

```diff
@@ -146,6 +145,7 @@ void ucc_tl_ucp_scatter_knomial_progress(ucc_coll_task_t *coll_task)
                     peer_seg_count * dt_size, mem_type, INV_VRANK(peer,
                     (ucc_rank_t)args->root, size), team, task), task, out);
             }
+            /*TODO: local_seg_index is always zero since rank that sends is base root? */
```

Comment:
is this TODO still a needed cooment?

Reply:
yes, i think this statement is true and we can clean code a bit